test/c/test_circuit.c

```diff
@@ -915,6 +916,107 @@ static int test_not_unitary_gate(void) {
     return result;
 }
 
+/**
+ * Test getting instruction parameters.
+ */
+static int test_get_instruction_params(void) {
+    QkCircuit *qc = qk_circuit_new(1, 0);
+    uint32_t q0[1] = {0};
+    double angle[1] = {0.23};
+
+    QkParam *theta = qk_param_new_symbol("theta");
+    double r_angle = 0.2345;
+    QkParam *val = qk_param_from_double(r_angle);
+    const QkParam *angles[2] = {theta, val};
+
+    QkComplex64 c0 = {0.0, 0.0};
+    QkComplex64 c1 = {1.0, 0.0};
+    QkComplex64 matrix[16] = {c1, c1, c0, c0,  // this
+                              c1, c1, c0, c0,  // is
+                              c0, c0, c1, c0,  // for
+                              c0, c0, c0, c1}; // formatting
+
+    int result = Ok;
+
+    if (qk_circuit_gate(qc, QkGate_SX, q0, NULL) != QkExitCode_Success) {
+        result = RuntimeError;
+        goto cleanup;
+    }
+    if (qk_circuit_gate(qc, QkGate_RX, q0, angle) != QkExitCode_Success) {
+        result = RuntimeError;
+        goto cleanup;
+    }
+    if (qk_circuit_gate_param(qc, QkGate_R, q0, angles) != QkExitCode_Success) {
+        result = RuntimeError;
+        goto cleanup;
+    }
+    if (qk_circuit_unitary(qc, matrix, q0, 1, false) != QkExitCode_Success) {
+        result = RuntimeError;
+        goto cleanup;
+    }
+
+    QkCircuitInstruction inst;
+
+    // SX has no parameters
+    qk_circuit_get_instruction(qc, 0, &inst);
+    if (inst.num_params != 0) {
+        printf("Expected 0 parameters in SX, got %u", inst.num_params);
+        result = EqualityError;
+        goto cleanup;
+    }
+
+    // RX has one parameter
+    qk_circuit_get_instruction(qc, 1, &inst);
+    if (inst.num_params != 1) {
+        printf("Expected 1 parameter in RX, got %u", inst.num_params);
+        result = EqualityError;
+        goto cleanup;
+    }
+
+    double rz_angle = qk_param_as_real(inst.params[0]);
+    if (fabs(rz_angle - angle[0]) > 1e-10) {
+        printf("Unexpected parameter value in RX gate");
+        result = EqualityError;
+        goto cleanup;
+    }
```

Comment:
If `rz_angle` is NaN because the parameter wasn't an `f64`, this will spuriously pass because `NaN` fails all arithmetic comparisons.

Shouldn't this equality be floating-point exact, anyway?

(Same comment in a couple of places.)